geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/SpatialUnits.groovy

```diff
@@ -351,53 +351,85 @@ IProcess createBlocks() {
+ * The selection of which polygon(s) need to be conserved is based on shared polygons area. By default, this parameter
+ * has no value and thus all relations are conserved
  *
- * @return A database table name and the name of its ID field
+ * @return outputTableName A table name containing ID from table 1, ID from table 2 and AREA shared by the two objects (if pointOnSurface = false)
+ * @return idColumnTab2 The ID name of the table 2
  */
-IProcess createScalesRelations() {
+IProcess spatialJoin() {
     return create {
-        title "Creating the Tables of relations between two scales"
-        id "createScalesRelations"
-        inputs inputLowerScaleTableName: String, inputUpperScaleTableName: String, idColumnUp: String,
-                prefixName: String, nbRelations: 1, datasource: JdbcDataSource
-        outputs outputTableName: String, outputIdColumnUp: String
-        run { inputLowerScaleTableName, inputUpperScaleTableName, idColumnUp, prefixName, nbRelations, datasource ->
+        title "Creating the spatial join between two tables of polygons"
+        id "spatialJoin"
+        inputs inputTable1: String, inputTable2: String, idColumnTab1: String,
+                idColumnTab2: String, prefixName: String, pointOnSurface: false, nbRelations: int,
+                datasource: JdbcDataSource
+        outputs outputTableName: String, idColumnTab2: String
+        run { inputTable1, inputTable2, idColumnTab1, idColumnTab2, prefixName, pointOnSurface,
+              nbRelations, datasource ->
 
             def GEOMETRIC_COLUMN_LOW = "the_geom"
             def GEOMETRIC_COLUMN_UP = "the_geom"
 
-            info "Creating the Tables of relations between the twos scales :  $inputLowerScaleTableName and $inputUpperScaleTableName"
+            info "Creating a spatial join between objects from two tables :  $inputTable1 and $inputTable2"
 
             // The name of the outputTableName is constructed (the prefix name is not added since it is already contained
             // in the inputLowerScaleTableName object
-            def outputTableName = postfix inputLowerScaleTableName, "corr"
-            datasource."$inputLowerScaleTableName".the_geom.createSpatialIndex()
-            datasource."$inputUpperScaleTableName".the_geom.createSpatialIndex()
+            def outputTableName = postfix "${inputTable1}_${inputTable2}", "join"
+            datasource."$inputTable1".the_geom.createIndex()
+            datasource."$inputTable2".the_geom.createIndex()
+            datasource."$inputTable2"[idColumnTab2].createIndex()
+
+            def query = """DROP TABLE IF EXISTS $outputTableName;"""
+            def subqueryNbRelations = ""
+            if (pointOnSurface){
+                query += """CREATE TABLE $outputTableName AS SELECT a.*, b.$idColumnTab2 
+                                    FROM $inputTable1 a, $inputTable2 b 
```

Comment:
CREATE TABLE $outputTableName AS SELECT a.*, b.$idColumnTab2 
                                    FROM $inputTable1 a, $inputTable2 b 
                                    WHERE   a.$GEOMETRIC_COLUMN_LOW && b.$GEOMETRIC_COLUMN_UP AND 
                                            ST_INTERSECTS(ST_POINTONSURFACE(a.$GEOMETRIC_COLUMN_LOW), st_force2d(b.$GEOMETRIC_COLUMN_UP))

Reply:
ok